### PR TITLE
test: integration test for setting up a social connector

### PR DESCRIPTION
**Pull request by IceHe**
## Summary

Integration tests: set up a social connector.

## Testing

Ditto

### CODE REVIEW

File: packages/integration-tests/tests/connectors.test.ts [hunk 1]
```diff
@@ -24,3 +31,39 @@ describe('list connectors and get details before setting up', () => {
...
+        clientSecret: 'secret_bar',
+      };
+
+      const updatedConfigResponse = await updateConnectorConfig(
+        facebookConnectorId,
+        facebookConfig
+      );
+      expect(updatedConfigResponse).toHaveProperty('statusCode', 200);
+      expect(() => {
+        const updatedConnectorDto = JSON.parse(updatedConfigResponse.body) as ConnectorDto;
```
Comment (gao-sun) [inline packages/integration-tests/tests/connectors.test.ts:49]: is this `as` necessary?

File: packages/integration-tests/src/connector-api.ts [hunk 2]
```diff
@@ -4,3 +4,21 @@ export const listConnectors = async () => authedAdminApi.get('connectors');
...
 export const getConnectorDetails = async (connectorId: string) =>
   authedAdminApi.get(`connectors/${connectorId}`);
+
+export const updateConnectorConfig = async (connectorId: string, config: Record<string, unknown>) =>
+  authedAdminApi.patch({
+    url: `connectors/${connectorId}`,
+    json: { config },
+    headers: {
+      'development-user-id': 'integration-test-admin-user',
+    },
```
[the remark this answers is not in the document]
Comment (IceHe) [reply packages/integration-tests/src/connector-api.ts:14]: should be removed.

File: packages/integration-tests/tests/connectors.test.ts (hunk 1 above)
Comment (IceHe) [reply packages/integration-tests/tests/connectors.test.ts]: Do it another way. PTAL

File: packages/integration-tests/tests/connectors.test.ts [hunk 3]
```diff
@@ -15,7 +22,7 @@ describe('list connectors and get details before setting up', () => {
   });
 
   it('should succeed to get connector details', async () => {
-    const response = await getConnectorDetails('facebook-universal');
+    const response = await getConnectorDetails(facebookConnectorId);
     expect(response).toHaveProperty('statusCode', 200);
     expect(() => {
       const connectorDto = JSON.parse(response.body) as ConnectorDto;
```
Comment (simeng-li) [inline packages/integration-tests/tests/connectors.test.ts:28]: user Zod to validate the response?

Comment (IceHe) [reply packages/integration-tests/tests/connectors.test.ts]: refactored. PTAL

**Comment by IceHe** [comment]
Rebased master to resolve conflicts.

**Review by gao-sun — APPROVED** [review]
cool

**Pull request merged**